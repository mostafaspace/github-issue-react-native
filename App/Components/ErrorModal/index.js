import React from 'react';
import {Modal, Text, View, TouchableWithoutFeedback} from 'react-native';
import {Button} from '../Button';
import Styles from './styles';
import PropTypes from 'prop-types';

export const ErrorModal = ({isVisible, hide, title, onPressOK, testID}) => {
    return (
        <Modal animationType="fade" transparent={true} visible={isVisible} onRequestClose={() => hide()} testID={testID ? testID : 'Modal'}>
            <TouchableWithoutFeedback onPress={() => hide()}>
                <View style={Styles.container}>
                    <View style={Styles.popUpContainer}>
                        <View style={Styles.headerContainer}>
                            <Text style={Styles.headerContainerText}>{'Error'}</Text>
                        </View>

                        <View style={Styles.messageContainer}>
                            <Text style={Styles.messageLabel}>{!title ? 'Something Went Wrong' : title}</Text>
                        </View>
                        <Button
                            text={'Ok'}
                            containerStyle={Styles.buttonContainer}
                            textStyle={Styles.buttonText}
                            onPress={() => {
                                hide();
                                onPressOK && onPressOK();
                            }}
                        />
                    </View>
                </View>
            </TouchableWithoutFeedback>
        </Modal>
    );
};

ErrorModal.propTypes = {
    title: PropTypes.string,
    isVisible: PropTypes.bool,
    onPressOK: PropTypes.func,
    hide: PropTypes.func,
};
