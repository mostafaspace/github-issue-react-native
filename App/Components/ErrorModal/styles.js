import {StyleSheet} from 'react-native';
import {Colors} from '../../Constants';

export default StyleSheet.create({
    container: {
        flex: 1,
        justifyContent: 'center',
        alignContent: 'center',
        alignItems: 'center',
        backgroundColor: 'rgba(20, 154, 222,0.8)',
    },
    popUpContainer: {
        width: '85%',
        height: null,
        backgroundColor: 'white',
        borderRadius: 30,
        flexDirection: 'column',
        alignItems: 'center',
        justifyContent: 'space-between',
    },
    headerContainer: {
        padding: 8,
        width: '100%',
        alignItems: 'center',
        borderTopLeftRadius: 30,
        borderTopRightRadius: 30,
    },
    headerContainerText: {
        fontSize: 10,
        fontWeight: 'bold',
        color: Colors.grey,
    },
    messageContainer: {
        margin: 16,
        alignItems: 'center',
    },
    messageLabel: {
        lineHeight: 24,
        marginHorizontal: 8,
        textAlign: 'center',
        fontWeight: 'bold',
        fontSize: 16,
        color: Colors.grey,
    },
    buttonContainer: {
        backgroundColor: Colors.lightGrey,
        width: '100%',
        margin: 0,
        flexDirection: 'row',
        justifyContent: 'center',
        borderTopLeftRadius: 0,
        borderTopRightRadius: 0,
        borderBottomLeftRadius: 30,
        borderBottomRightRadius: 30,
    },
    buttonText: {
        color: Colors.grey,
        fontWeight: 'bold',
    },
});
