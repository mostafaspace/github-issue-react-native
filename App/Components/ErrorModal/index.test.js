import React from 'react';
import {render, cleanup} from '@testing-library/react-native';
import {ErrorModal} from '../ErrorModal';
import renderer from 'react-test-renderer';
import '@testing-library/jest-native/extend-expect';

afterEach(cleanup);
it('renders component', () => {
    render(<ErrorModal />);
});

it('check if has text', () => {
    const {queryByTestId, getByText} = render(<ErrorModal title={'title text'} />);
    expect(queryByTestId('Modal')).toContainElement(getByText('title text'));
});

it('ErrorModal matches snapshot', () => {
    const tree = renderer.create(<ErrorModal text={'test'} />).toJSON()
    expect(tree).toMatchSnapshot();
});
