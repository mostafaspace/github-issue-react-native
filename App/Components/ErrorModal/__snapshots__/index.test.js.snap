// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorModal matches snapshot 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  testID="Modal"
  transparent={true}
  visible={true}
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignContent": "center",
        "alignItems": "center",
        "backgroundColor": "rgba(20, 154, 222,0.8)",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 30,
          "flexDirection": "column",
          "height": null,
          "justifyContent": "space-between",
          "width": "85%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "borderTopLeftRadius": 30,
            "borderTopRightRadius": 30,
            "padding": 8,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#575757",
              "fontSize": 10,
              "fontWeight": "bold",
            }
          }
        >
          Error
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "margin": 16,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#575757",
              "fontSize": 16,
              "fontWeight": "bold",
              "lineHeight": 24,
              "marginHorizontal": 8,
              "textAlign": "center",
            }
          }
        >
          Something Went Wrong
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#dddede",
            "borderBottomLeftRadius": 30,
            "borderBottomRightRadius": 30,
            "borderRadius": 16,
            "borderTopLeftRadius": 0,
            "borderTopRightRadius": 0,
            "elevation": 1,
            "flexDirection": "row",
            "justifyContent": "center",
            "margin": 0,
            "opacity": 1,
            "paddingHorizontal": 16,
            "paddingVertical": 16,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 0.18,
            "shadowRadius": 1,
            "width": "100%",
          }
        }
        testID="Button"
      >
        <Text
          numberOfLines={1}
          style={
            Object {
              "alignSelf": "center",
              "color": "#ffffff",
              "flex": 1,
              "paddingHorizontal": 16,
              "textAlign": "center",
            }
          }
          testID="ButtonText"
        >
          Ok
        </Text>
      </View>
    </View>
  </View>
</Modal>
`;
